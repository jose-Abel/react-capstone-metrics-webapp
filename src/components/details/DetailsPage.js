import React from 'react';

const DetailsPage = () => {
  const dummydata = [];

  return (
    <div>
      {dummydata}
    </div>
  );
};

export default DetailsPage;
