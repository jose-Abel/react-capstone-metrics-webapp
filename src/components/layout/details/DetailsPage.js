import React from 'react';
import { useParams } from 'react-router';
import { useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import classes from './DetailsPage.module.css';

const DetailsPage = () => {
  const params = useParams();
  const cryptos = useSelector((state) => state);
  const crypto = cryptos.filter((cryp) => cryp.id === params.id)[0];

  const navigate = useNavigate();

  const handleClickIcon = () => {
    const path = '/home';
    navigate(path);
  };

  return (
    <section>
      <i
        className={`${'fas fa-arrow-circle-left'} ${classes.fontawesomeIcon}`}
        onClick={handleClickIcon}
        onKeyDown={handleClickIcon}
        role="button"
        tabIndex={0}
        aria-label="Back button"
      />
      <header className={classes.detailsHeader}>
        <img src={crypto.image} alt={crypto.name} />
        <p>
          {crypto.name}
        </p>
      </header>
      <div className={classes.card}>
        <p className={classes.rowContainer}>
          <span className={classes.rowText}>
            All Time High
          </span>
          {' '}
          {crypto.ath}
        </p>
        <p className={classes.rowContainer}>
          <span className={classes.rowText}>
            All Time High
          </span>
          {' '}
          {crypto.ath}
        </p>
        <p className={classes.rowContainer}>
          <span className={classes.rowText}>
            All Time High Change Percentage
          </span>
          {' '}
          {crypto.ath_change_percentage}
        </p>

        <p className={classes.rowContainer}>
          <span className={classes.rowText}>
            Circulating Supply
          </span>
          {' '}
          {crypto.circulating_supply}
        </p>

        <p className={classes.rowContainer}>
          <span className={classes.rowText}>
            Current Price
          </span>
          {' '}
          {crypto.current_price}
        </p>

        <p className={classes.rowContainer}>
          <span className={classes.rowText}>
            Highest Price last 24 hours
          </span>
          {' '}
          {crypto.high_24h}
        </p>

        <p className={classes.rowContainer}>
          <span className={classes.rowText}>
            Lowest Price last 24 hours
          </span>
          {' '}
          {crypto.low_24h}
        </p>

        <p className={classes.rowContainer}>
          <span className={classes.rowText}>
            Market Cap Rank:
          </span>
          {' '}
          {crypto.market_cap_rank}
        </p>

      </div>
    </section>
  );
};

export default DetailsPage;

/*
ath: 686.31
ath_change_percentage: -14.1262
ath_date: "2021-05-10T07:24:17.097Z"
atl: 0.0398177
atl_change_percentage: 1480039.68197
atl_date: "2017-10-19T00:00:00.000Z"
circulating_supply: 168137035.9
current_price: 587.97
fully_diluted_valuation: 98939660242
high_24h: 639.24
id: "binancecoin"
image: "https://assets.coingecko.com/coins/images/825/large/binance-coin-logo.png?1547034615"
last_updated: "2021-11-16T22:04:43.927Z"
low_24h: 575.88
market_cap: 98939660242
market_cap_change_24h: -8245621072.922806
market_cap_change_percentage_24h: -7.69287
market_cap_rank: 3
max_supply: 168137035.9
name: "Binance Coin"
price_change_24h: -47.887937631762
price_change_percentage_24h: -7.53125
roi: null
symbol: "bnb"
total_supply: 168137035.9
total_volume: 2961108361

*/
