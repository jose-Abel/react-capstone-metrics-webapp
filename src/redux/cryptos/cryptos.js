const GET_CRYPTOS_LIST_SUCCESS = 'react-capstone-metrics-webapp/cryptos/GET_CRYPTOS_LIST_SUCCESS';
const GET_CRYPTOS_LIST_FAILURE = 'react-capstone-metrics-webapp/cryptos/GET_CRYPTOS_LIST_FAILURE';

const initialState = [];

const baseURL = 'https://api.coingecko.com/api/v3/coins/markets?vs_currency=usd&order=market_cap_desc&per_page=100&page=1&sparkline=false';

export const getCryptosList = () => async (dispatch) => {
  try {
    const response = await fetch(`${baseURL}`);
    const data = await response.json();

    dispatch({ type: GET_CRYPTOS_LIST_SUCCESS, payload: data });
  } catch (err) {
    dispatch({ type: GET_CRYPTOS_LIST_FAILURE, err });
  }
};

export default (state = initialState, action) => {
  switch (action.type) {
    case GET_CRYPTOS_LIST_SUCCESS:
      return [
        ...action.payload,
      ];

    default:
      return state;
  }
};
