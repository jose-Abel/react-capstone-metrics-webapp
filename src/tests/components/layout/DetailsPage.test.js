/* eslint-disable no-labels */
/* eslint-disable no-restricted-syntax */
import React from 'react';
import { render, screen } from '@testing-library/react';
import * as Redux from 'react-redux';
import * as RouterDOM from 'react-router-dom';
import store from '../../../redux/configureStore';
import '@testing-library/jest-dom/extend-expect';
import DetailsPage from '../../../components/layout/details/DetailsPage';
import cryptos from '../../fixtures/cryptos';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: jest.fn(),
}));

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn(),
}));

describe('DetailsPage', () => {
  const useParamsMock = RouterDOM.useParams;
  const useSelectorMock = Redux.useSelector;

  beforeEach(() => {
    useParamsMock.mockImplementation(() => ({ id: cryptos[0].id }));
    useSelectorMock.mockImplementation((selector) => selector(cryptos));
  });

  test('renders DetailsPage component correctly', () => {
    const tree = render(
      <Redux.Provider store={store}>
        <RouterDOM.BrowserRouter>
          <DetailsPage />
        </RouterDOM.BrowserRouter>
      </Redux.Provider>,
    );
    expect(tree).toMatchSnapshot();
  });

  test('find text in the DetailsPage component DOM', () => {
    render(
      <Redux.Provider store={store}>
        <RouterDOM.BrowserRouter>
          <DetailsPage />
        </RouterDOM.BrowserRouter>
      </Redux.Provider>,
    );

    expect(screen.getByText(/All Time High:/)).toBeInTheDocument();
  });
});
