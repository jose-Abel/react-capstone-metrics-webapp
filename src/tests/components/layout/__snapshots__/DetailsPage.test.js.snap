// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsPage renders DetailsPage component correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section>
        <i
          aria-label="Back button"
          class="fas fa-arrow-circle-left fontawesomeIcon"
          role="button"
          tabindex="0"
        />
        <header
          class="detailsHeader"
        >
          <img
            alt="Bitcoin"
            class="detailsImage"
            src="https://assets.coingecko.com/coins/images/1/large/bitcoin.png?1547033579"
          />
          <p>
            Bitcoin
          </p>
        </header>
        <div
          class="card"
        >
          <p
            class="rowContainer"
          >
            <span
              class="rowText"
            >
              All Time High:
            </span>
             
            69045
          </p>
          <p
            class="rowContainer"
          >
            <span
              class="rowText"
            >
              All Time High Change Percentage:
            </span>
             
            -12.27579
          </p>
          <p
            class="rowContainer"
          >
            <span
              class="rowText"
            >
              Circulating Supply:
            </span>
             
            18875325
          </p>
          <p
            class="rowContainer"
          >
            <span
              class="rowText"
            >
              Current Price:
            </span>
             
            60579
          </p>
          <p
            class="rowContainer"
          >
            <span
              class="rowText"
            >
              Highest Price last 24 hours:
            </span>
             
            64042
          </p>
          <p
            class="rowContainer"
          >
            <span
              class="rowText"
            >
              Lowest Price last 24 hours:
            </span>
             
            59150
          </p>
          <p
            class="rowContainer"
          >
            <span
              class="rowText"
            >
              Market Cap Rank:
            </span>
             
            1
          </p>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section>
      <i
        aria-label="Back button"
        class="fas fa-arrow-circle-left fontawesomeIcon"
        role="button"
        tabindex="0"
      />
      <header
        class="detailsHeader"
      >
        <img
          alt="Bitcoin"
          class="detailsImage"
          src="https://assets.coingecko.com/coins/images/1/large/bitcoin.png?1547033579"
        />
        <p>
          Bitcoin
        </p>
      </header>
      <div
        class="card"
      >
        <p
          class="rowContainer"
        >
          <span
            class="rowText"
          >
            All Time High:
          </span>
           
          69045
        </p>
        <p
          class="rowContainer"
        >
          <span
            class="rowText"
          >
            All Time High Change Percentage:
          </span>
           
          -12.27579
        </p>
        <p
          class="rowContainer"
        >
          <span
            class="rowText"
          >
            Circulating Supply:
          </span>
           
          18875325
        </p>
        <p
          class="rowContainer"
        >
          <span
            class="rowText"
          >
            Current Price:
          </span>
           
          60579
        </p>
        <p
          class="rowContainer"
        >
          <span
            class="rowText"
          >
            Highest Price last 24 hours:
          </span>
           
          64042
        </p>
        <p
          class="rowContainer"
        >
          <span
            class="rowText"
          >
            Lowest Price last 24 hours:
          </span>
           
          59150
        </p>
        <p
          class="rowContainer"
        >
          <span
            class="rowText"
          >
            Market Cap Rank:
          </span>
           
          1
        </p>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
